### ISSUE TITLE
[Bug]: SEHException is thrown at Realms.SynchronizationContextScheduler.scheduler_invoke_function

### ISSUE THREAD

**Issue by markawhite-broadcom** [realm/realm-dotnet#2824]
### What happened?

This exception happens over 90% of the time when I run a my application using v10.9.0, about 20% of the time with v10.8.0, and has not happened with v10.7.1.  My application is running multiple tasks (8 at a time).  Each task consumes a large amount of data from  the system.   Data examples are file/directory listings, registry values, windows events, etc.  The data is pushed into a Realm database from the different tasks/threads.  The issue happens at some point while my application is capturing the data.  My application hooks the AppDomain.CurrentDomain.UnhandledException and records this exception before it exits:

SymDiagWin.App.CurrentDomain_UnhandledException	Exception: System.Runtime.InteropServices.SEHException (0x80004005): External component has thrown an exception.
   at Realms.SynchronizationContextScheduler.scheduler_invoke_function(IntPtr function_ptr, Boolean can_execute)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()

The rest of the threads continue to run and collect data.  My application logs the exception and notifies the user of the exception.  As soon as the Ok button is clicked, my application closes.

### Repro steps

Run my application which is using a combination of .Net Standard v2.0 and .Net 4.6.2 Framework code

### Version

4.6.2

### What SDK flavour are you using?

Local Database only

### What type of application is this?

WPF

### Client OS and version

Windows 10 Pro 21H2 19044.1526

### Stacktrace of the exception/crash you're getting

```shell
SymDiagWin.App.CurrentDomain_UnhandledException	Exception: System.Runtime.InteropServices.SEHException (0x80004005): External component has thrown an exception.
   at Realms.SynchronizationContextScheduler.scheduler_invoke_function(IntPtr function_ptr, Boolean can_execute)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

**Comment by LaPeste** [comment]
Hi, thank you for taking the time to report the issue.
This could be bug, but we need a repro project to be able to investigate it further. Could you try extrapolating the issue in a  project send it to us?
Otherwise, in case it's possible for you to share the full project with us then we can supply you with an invitation to upload your project in our internal and secure cloud storage.
Let us know which one could be the preferable way for you.

**Comment by LaPeste** [comment]
It's been a while since any update for this issue were posted. Hence, I'm closing it.
However, @markawhite-broadcom feel free to reopen it if you get more info about it.

**Comment by markawhite-broadcom** [comment]
While doing a code review I found the issue.  My application uses a thread to check the database to see if a cmd should run, and to also update the cmd with progress information.  The cmd is run on a different thread while storing information in the database.  The issue was that each time the cmd was updated in the database, it used the original cmd object.  I changed the code to retrieve the cmd object each time before updating it.  With v10.11.0 I am not seeing the exception.
